src/sign-typed-data.test.ts

```diff
@@ -1,17 +1,232 @@
 import * as ethUtil from 'ethereumjs-util';
+import Ajv from 'ajv';
 import {
   recoverTypedSignature,
   signTypedData,
   TypedDataUtils,
   typedSignatureHash,
   SignTypedDataVersion,
+  TYPED_MESSAGE_SCHEMA,
 } from './sign-typed-data';
 
 const privateKey = Buffer.from(
   '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0',
   'hex',
 );
 
+/**
+ * Get a list of all Solidity types supported by EIP-712.
+ *
+ * @returns A list of all supported Solidity types.
+ */
+function getEip712SolidityTypes() {
+  const types = ['bool', 'address', 'string', 'bytes'];
+  const ints = Array.from(new Array(32)).map(
+    (_, index) => `int${(index + 1) * 8}`,
+  );
+  const uints = Array.from(new Array(32)).map(
+    (_, index) => `uint${(index + 1) * 8}`,
+  );
+  const bytes = Array.from(new Array(32)).map(
+    (_, index) => `bytes${index + 1}`,
+  );
+
+  return [...types, ...ints, ...uints, ...bytes];
+}
+
+const eip712SolidityTypes = getEip712SolidityTypes();
+
+describe('TYPED_MESSAGE_SCHEMA', () => {
+  it('should match valid typed message', () => {
+    const ajv = new Ajv();
+    const validate = ajv.compile(TYPED_MESSAGE_SCHEMA);
+    const typedMessage = {
+      domain: {},
+      message: {},
+      primaryType: 'object',
+      types: {
+        EIP712Domain: [],
+      },
+    };
+
+    expect(validate(typedMessage)).toBe(true);
+  });
+
+  it('should allow custom types in addition to domain', () => {
+    const ajv = new Ajv();
+    const validate = ajv.compile(TYPED_MESSAGE_SCHEMA);
+    const typedMessage = {
+      domain: {},
+      message: {},
+      primaryType: 'Message',
+      types: {
+        EIP712Domain: [],
+        Message: [],
+      },
+    };
+
+    expect(validate(typedMessage)).toBe(true);
+  });
+
+  for (const solidityType of eip712SolidityTypes) {
+    // eslint-disable-next-line no-loop-func
```

Comment:
[nit] If you use `.forEach` instead of a `for` loop, does it remove the need for this override?

Reply:
It does, but now that you've brought this up, I kinda want to get to the bottom of this. This ought to work. Something is messed up with the ESLint globals configuration. I'll look into this in a subsequent PR.

Reply:
Oh my goodness. I found the problem. The override for the test ESLint config is wrong, the file glob uses the `js` file extension rather than `ts`. There are a bunch of violations, I'll fix this separately.